backend/state.py

```diff
@@ -224,18 +223,16 @@ def update_special_effect(self, special_effect, arg, param_arg_dict):
         Args:
             special_effect (SpecialEffect): The special effect to update.
             arg (Object): The object to update the special effect for.
-            param_arg_dict (Dictionary[Object, Object]): The arguments for the 
+            param_arg_dict (Dictionary[Str, Object]): The arguments for the 
                 special effect.
         """
-        if not special_effect.arg:
-            new_special_effect = special_effect.apply_sfx_on_arg(arg, param_arg_dict)
-            self.special_effects.append(new_special_effect)
-            current = self.special_effects[self.special_effects.index(new_special_effect)]
-        else:
-            current = self.special_effects[self.special_effects.index(special_effect)]
+        replaced_effect = special_effect.apply_sfx_on_arg(arg, param_arg_dict)
+        if replaced_effect in self.special_effects:
+            return
+        if not replaced_effect in self.special_effects:
```

Comment:
there is no need for this if statement, in the case that `if not replaced_effect in self.special_effects` isn't True it would've returned from the previous if statement.

Reply:
removed if statement